core/client/fs/src/main/java/alluxio/client/file/MetadataCachingBaseFileSystem.java

```diff
@@ -12,128 +12,53 @@
 package alluxio.client.file;
 
 import alluxio.AlluxioURI;
-import alluxio.conf.AlluxioConfiguration;
-import alluxio.conf.PropertyKey;
 import alluxio.exception.AlluxioException;
 import alluxio.exception.FileDoesNotExistException;
-import alluxio.grpc.Bits;
 import alluxio.grpc.GetStatusPOptions;
 import alluxio.grpc.ListStatusPOptions;
-import alluxio.util.FileSystemOptions;
-import alluxio.util.ThreadUtils;
 
-import com.google.common.annotations.VisibleForTesting;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.google.common.io.Closer;
 
 import java.io.IOException;
 import java.util.List;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.RejectedExecutionException;
-import java.util.concurrent.SynchronousQueue;
-import java.util.concurrent.ThreadPoolExecutor;
-import java.util.concurrent.TimeUnit;
-
-import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * FileSystem implementation with the capability of caching metadata of paths.
  */
-@ThreadSafe
 public class MetadataCachingBaseFileSystem extends BaseFileSystem {
-  private static final Logger LOG = LoggerFactory.getLogger(BaseFileSystem.class);
-  private static final int THREAD_KEEPALIVE_SECOND = 60;
-  private static final int THREAD_TERMINATION_TIMEOUT_MS = 10000;
-
-  private final MetadataCache mMetadataCache;
-  private final ExecutorService mAccessTimeUpdater;
+  private final MetadataCacheManager mMetadataCacheManager;
+  private final FileSystemContext mFsContext;
+  private final Closer mCloser;
 
   /**
+   * @param fs delegated file system
    * @param context the fs context
    */
-  public MetadataCachingBaseFileSystem(FileSystemContext context) {
+  public MetadataCachingBaseFileSystem(FileSystem fs, FileSystemContext context) {
```

Comment:
Should this be a delegating file system as well? Although we pass in `fs`, there are two filesystems being invoked, one that is passed in for calls to `MetadataCacheManager` and another for most operations (fall back to `BaseFileSystem` implementation).